### PR TITLE
7903857: apidiff: Include test run in GitHub Actions workflow

### PR DESCRIPTION
**Pull request by sormuras**
Please review this change to actually run ApiDiff's self-tests on all three major platforms.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Integration blocker
&nbsp;⚠️ Title mismatch between PR and JBS for issue [CODETOOLS-7903857](https://bugs.openjdk.org/browse/CODETOOLS-7903857)

### Issue
 * [CODETOOLS-7903857](https://bugs.openjdk.org/browse/CODETOOLS-7903857): Include test run in GitHub Actions workflow (**Enhancement** - P4) ⚠️ Title mismatch between PR and JBS.

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/apidiff.git pull/23/head:pull/23` \
`$ git checkout pull/23`

Update a local copy of the PR: \
`$ git checkout pull/23` \
`$ git pull https://git.openjdk.org/apidiff.git pull/23/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 23`

View PR using the GUI difftool: \
`$ git pr show -t 23`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/apidiff/pull/23.diff">https://git.openjdk.org/apidiff/pull/23.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/apidiff/pull/23#issuecomment-2387875019)

### CODE REVIEW

**Comment by pavelrappo** [comment]
How did you verify that the updated workflow works?

**Comment by sormuras** [comment]
> How did you verify that the updated workflow works?

They did not! See latest failed runs at https://github.com/sormuras/apidiff/actions/workflows/test.yml

File: .github/workflows/test-linux.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: 'Test (linux)'
+
+on:
+  workflow_call:
+
+jobs:
+  test:
+    name: 'ApiDiff'
```
Comment (pavelrappo) [inline .github/workflows/test-linux.yml:8]: From what I can tell, Jon, the original author, prefers the following casing: APIDiff.

Comment (sormuras) [reply .github/workflows/test-linux.yml:8]: You're right. Will update all occurances in a follow-up PR.

**Pull request closed without merging**